code/game/objects/items/weapons/surgery_tools.dm

```diff
@@ -128,24 +151,60 @@
 
 
 /obj/item/weapon/scalpel/laser
-	heat_production = 0
+	icon_state = "stabslash"
 	damtype = "fire"
 	sharpness_flags = SHARP_TIP | SHARP_BLADE | HOT_EDGE
 	var/cauterymode = 0 //1 = cautery enabled
+	var/obj/item/weapon/scalpel/laser/held
+
+/obj/item/weapon/scalpel/laser/New()
+	..()
+	icon_state = "scalpel"
+	held = new /obj/item/weapon/cautery/laser(src)
 
 /obj/item/weapon/scalpel/laser/attack_self(mob/user)
-	if(!cauterymode)
+	if(!cauterymode && held)
 		to_chat(user, "You disable the blade and switch to the scalpel's cautery tool.")
 		heat_production = 1600
 		sharpness = 0
 		sharpness_flags = 0
+	else if(!held)
+		to_chat(user, "\The [src] lacks a cautery attachment.")
+		return
 	else
 		to_chat(user, "You return the scalpel to cutting mode.")
 		heat_production = 0
 		sharpness = initial(sharpness)
 		sharpness_flags = initial(sharpness_flags)
 	cauterymode = !cauterymode
 
+/obj/item/weapon/scalpel/laser/examine(mob/user)
+	..()
+	if(!cauterymode)
+		to_chat(user, "\The [src] is in cutting mode.")
+	else
+		to_chat(user, "\The [src] is in cautery mode.")
+
+/obj/item/weapon/scalpel/laser/attackby(var/obj/item/used_item, mob/user)
+	if(isscrewdriver(used_item) && cauterymode)
+		if(held)
+			to_chat(user, "<span class='notice'>You detach \the [held] and \the [src] switches to cutting mode.</span>")
+			playsound(get_turf(src), "sound/items/screwdriver.ogg", 10, 1)
+			held.add_fingerprint(user)
+			held.forceMove(get_turf(src))
+			held = null
+			heat_production = 0
+			sharpness = initial(sharpness)
+			sharpness_flags = initial(sharpness_flags)
+			cauterymode = 0
+	else if(istype(used_item, /obj/item/weapon/cautery/laser))
+		if(!held)
+			to_chat(user, "<span class='notice'>You attach \the [used_item] into \the [src].</span>")
+			playsound(get_turf(src), "sound/items/screwdriver.ogg", 10, 1)
+			user.drop_item(used_item, src)
```

Comment:
If the cautery is superglued to the mob's hand, or the mob is otherwise incapable of dropping it, the mob will end up with a cautery in their hand which the scalpel still has a reference to.
You need to account for the possibility that the mob cannot drop the item, using `if(user.drop_item(used_item, src))`. `drop_item()` returns a value that indicates whether it succeeded or failed.